week_1_2.py

```diff
@@ -0,0 +1,100 @@
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code.
+Feel free to add any more new features to your class.
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student.
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = dict()
+
+    def add_subject(self, subject, grade = "N/A"):
+        # Updating dictionary with subject and grade. If no grade is inputted, the grade is N/A instead.
+        self.subjects.update({subject: grade})
+
+    def remove_subject(self, subject):
+        self.subjects.pop(subject)
+
+    def view_subjects(self):
+        print("Subjects taken: " + str(list(self.subjects.keys())))
+
+    def view_subject_details(self):
+        print("Subjects and grades: " + str(self.subjects))
+
+    def get_average(self):
+        # Check if the inputted grades are numbers.
+        # Add the (numbered) grades up and divide it by the number of (numbered) grades to return an average.
+
+        all_grades = list(self.subjects.values())
+        no_grades = 0
+        total_grade = 0
+
+        for grade in all_grades:
+            if type(grade) == int or type(grade) == float:
+                no_grades +=1
+                total_grade += grade
+
+        average_grade = total_grade/no_grades
+
+        return average_grade
+
+    def view_average(self):
+        print("Average grade: " + str(self.get_average()))
+
+    def view_student_details(self):
```

Comment:
Can you remember the 'magic' (i.e. double underscore) method for formatting how an object is represented when printed?